File: engine/engine-model/src/main/scala/tools/packet/npctalk/AskSpeedQuiz.scala
package tools.packet.npctalk

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class AskSpeedQuiz(private var _speakerTypeId: Int, private var _speakerTemplateId: Int, private var _resCode: Int,
                   private var _type: Int, private var _answer: Int, private var _correct: Int,
                   private var _remain: Int, private var _remainInitialQuiz: Int) extends PacketInput {
  def speakerTypeId: Int = _speakerTypeId

  def speakerTemplateId: Int = _speakerTemplateId

  def resCode: Int = _resCode

  def theType: Int = _type

  def answer: Int = _answer

  def correct: Int = _correct

  def remain: Int = _remain

  def remainInitialQuiz: Int = _remainInitialQuiz

  override def opcode(): SendOpcode = SendOpcode.NPC_TALK
}

File: engine/engine-model/src/main/java/tools/packet/npctalk/AskSpeedQuiz.java
package tools.packet.npctalk;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record AskSpeedQuiz(Integer speakerTypeId, Integer speakerTemplateId, Integer resCode, Integer theType,
                           Integer answer, Integer correct, Integer remain,
                           Integer remainInitialQuiz) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.NPC_TALK;
   }
}